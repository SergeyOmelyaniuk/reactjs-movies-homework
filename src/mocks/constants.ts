import { Category, Genre, Language } from '../types';

export const categories: Category[] = [
	{ id: 1, value: 'popular', title: 'Popular' },
	{ id: 2, value: 'top_rated', title: 'Top rated' },
	{ id: 3, value: 'upcoming', title: 'Upcoming' },
];

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const genres: Genre[] = [
	{ id: 1, name: 'Genre 1' },
	{ id: 2, name: 'Genre 2' },
	{ id: 3, name: 'Genre 3' },
	{ id: 4, name: 'Genre 4' },
	{ id: 5, name: 'Genre 5' },
];
