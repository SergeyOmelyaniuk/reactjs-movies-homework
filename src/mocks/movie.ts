import { MovieDetails } from './../types';

export const movieDescription: MovieDetails = {
	poster_path: 'film_1.jpg',
	genres: [{ id: 1, name: 'first genre' }],
	title: 'Красотка на взводе',
	release_date: '2021-05-26',
	vote_average: 68,
	revenue: 120000,
	runtime: 120,
	overview:
		'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
};
