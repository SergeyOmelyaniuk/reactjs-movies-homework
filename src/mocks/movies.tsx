import { Movie } from '../types';

export const movies: Movie[] = [
	{
		id: 1,
		title: 'Красотка на взводе',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_1.jpg',
		vote_average: 68,
		popularity: 0.5,
		revenue: 120000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [
			{
				name: 'Kate Beckinsale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_1.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Lindy',
			},
			{
				name: 'Stanley Tucci',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_2.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Dr. Ivan Munchin',
			},
			{
				name: 'Bobby Cannavale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_3.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Vicars',
			},
			{
				name: 'Jai Courtney',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_4.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Justin',
			},
			{
				name: 'Laverne Cox',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_5.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Nevin',
			},
			{
				name: 'David Bradley',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_6.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Gareth Fizel',
			},
		],
	},
	{
		id: 2,
		title: 'Сокровище Счастливчика',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_2.jpg',
		vote_average: 72,
		popularity: 0.5,
		revenue: 230000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 3,
		title: 'Сталинград',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_3.jpg',
		vote_average: 76,
		popularity: 0.5,
		revenue: 240000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 4,
		title: 'Люди в черном: Интернэшнл',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_4.jpg',
		vote_average: 69,
		popularity: 0.5,
		revenue: 136000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 5,
		title: 'Значит, война',
		release_date: '2021-05-26',
		genre: ['Боевик'],
		img: 'film_5.jpg',
		vote_average: 96,
		popularity: 0.5,
		revenue: 226000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
];
