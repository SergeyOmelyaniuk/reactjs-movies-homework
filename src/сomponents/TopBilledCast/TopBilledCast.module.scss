.header {
	display: flex;
	justify-content: space-between;
}

.title {
	font-size: 28px;
}

.button {
	background-color: #151515;
	color: #fff;
	padding: 6px 9px;
	border: 1px solid #fff;
	border-radius: 5px;
	cursor: pointer;
	&:hover {
		background-color: #fff;
		color: #000;
	}
}

.wrap {
	display: flex;
	justify-content: space-between;
	padding-top: 30px;
}
