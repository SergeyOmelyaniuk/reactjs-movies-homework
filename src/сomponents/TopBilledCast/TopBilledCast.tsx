import React from 'react';
import ActorCard from '../ActorCard';
import styles from './TopBilledCast.module.scss';
import { Actor } from '../../types';

interface TopBilledCastProps {
	actors: Actor[];
}

const TopBilledCast = ({ actors }: TopBilledCastProps) => (
	<div className={styles.topBilledCast}>
		<div className={styles.header}>
			<span className={styles.title}>Top billed cast</span>
			<button className={styles.button}>Show all</button>
		</div>
		<div className={styles.wrap}>
			{actors.map((actor) => (
				<ActorCard key={actor.name} actor={actor} />
			))}
		</div>
	</div>
);

export default TopBilledCast;
