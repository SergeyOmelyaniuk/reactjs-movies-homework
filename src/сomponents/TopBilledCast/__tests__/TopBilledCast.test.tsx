import React from 'react';
import { render } from '@testing-library/react';
import TopBilledCast from '..';
import { movies } from '../../../mocks/movies';

describe('TopBilledCast', () => {
	test('should render', () => {
		const { container } = render(
			<TopBilledCast actors={movies.popular[0].actors} />
		);

		expect(container).toMatchSnapshot();
	});
});
