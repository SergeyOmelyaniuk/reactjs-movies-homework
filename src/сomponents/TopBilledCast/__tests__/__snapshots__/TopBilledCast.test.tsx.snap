// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBilledCast should render 1`] = `
<div>
  <div
    class="topBilledCast"
  >
    <div
      class="header"
    >
      <span
        class="title"
      >
        Top billed cast
      </span>
      <button
        class="button"
      >
        Show all
      </button>
    </div>
    <div
      class="wrap"
    >
      <div
        class="actorCard"
      >
        <div
          class="wrapImage"
        >
          <img
            alt=""
            class="image"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Kate Beckinsale
          </h2>
          <p
            class="character"
          >
            Lindy
          </p>
        </div>
      </div>
      <div
        class="actorCard"
      >
        <div
          class="wrapImage"
        >
          <img
            alt=""
            class="image"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Stanley Tucci
          </h2>
          <p
            class="character"
          >
            Dr. Ivan Munchin
          </p>
        </div>
      </div>
      <div
        class="actorCard"
      >
        <div
          class="wrapImage"
        >
          <img
            alt=""
            class="image"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Bobby Cannavale
          </h2>
          <p
            class="character"
          >
            Detective Vicars
          </p>
        </div>
      </div>
      <div
        class="actorCard"
      >
        <div
          class="wrapImage"
        >
          <img
            alt=""
            class="image"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Jai Courtney
          </h2>
          <p
            class="character"
          >
            Justin
          </p>
        </div>
      </div>
      <div
        class="actorCard"
      >
        <div
          class="wrapImage"
        >
          <img
            alt=""
            class="image"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Laverne Cox
          </h2>
          <p
            class="character"
          >
            Detective Nevin
          </p>
        </div>
      </div>
      <div
        class="actorCard"
      >
        <div
          class="wrapImage"
        >
          <img
            alt=""
            class="image"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            David Bradley
          </h2>
          <p
            class="character"
          >
            Gareth Fizel
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
